crates/parser/src/grammar/items.rs

```diff
@@ -59,13 +59,7 @@ pub(super) fn item_or_macro(p: &mut Parser<'_>, stop_on_r_curly: bool) {
     };
 
     if paths::is_use_path_start(p) {
-        match macro_call(p) {
-            BlockLike::Block => (),
-            BlockLike::NotBlock => {
-                p.expect(T![;]);
-            }
-        }
-        m.complete(p, MACRO_CALL);
```

Comment:
So, everything the parser eats before a `complete` call will be attached to the `completed` node. The reason we eat the semicolon here, instead of constructing the macro call in the `macro_call` function is because that functio nis used for parsing a macro call as an item, or as an expression. The semicolon here is only valid when parsed as an item though, hence why the completion is handled in this weird way. So we want to keep things this way here. This is an important distinction as it changes the parse for stuff like `let foo = macro_rules!();`, the semi here belongs to the let statement, not the macro call.